staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -273,20 +287,43 @@ func (s *SpdyRoundTripper) tlsConn(ctx context.Context, rwc net.Conn, targetHost
 // dialWithoutProxy dials the host specified by url, using TLS if appropriate.
 func (s *SpdyRoundTripper) dialWithoutProxy(ctx context.Context, url *url.URL) (net.Conn, error) {
 	dialAddr := netutil.CanonicalAddr(url)
-	dialer := s.Dialer
-	if dialer == nil {
-		dialer = &net.Dialer{}
+	dial := s.DialContext
+	if dial == nil && s.Dialer != nil {
+		dial = s.Dialer.DialContext
+	} else if dial == nil {
+		nd := &net.Dialer{}
+		dial = nd.DialContext
 	}
 
+	conn, err := dial(ctx, "tcp", dialAddr)
+	if err != nil {
+		return nil, err
+	}
 	if url.Scheme == "http" {
-		return dialer.DialContext(ctx, "tcp", dialAddr)
+		return conn, nil
 	}
 
-	tlsDialer := tls.Dialer{
```

Comment:
so, we basically copy the `dial` code from the tls stdlibrary https://cs.opensource.google/go/go/+/refs/tags/go1.19.2:src/crypto/tls/tls.go;l=120;drc=fdea8e21708e9ed04498a7f7b68cbee43005d58d

Reply:
Yeah, I added a reference.